sdk/src/akkaserverless.js

```diff
@@ -183,13 +187,24 @@ class AkkaServerless {
 
     this.server.addService(discovery, {
       discover: this.discover.bind(this),
-      reportError: this.reportError.bind(this)
+      reportError: this.reportError.bind(this),
+      proxyTerminated: this.proxyTerminated.bind(this)
     });
 
     const boundPort = this.server.bind(opts.bindAddress + ":" + opts.bindPort, grpc.ServerCredentials.createInsecure());
     this.server.start();
     console.log("gRPC server started on " + opts.bindAddress + ":" + boundPort);
 
+    process.on('SIGTERM', function onSigterm () {
+      if (!this.proxySeen || this.proxyHasTerminated || this.devMode) {
+        debug('Got SIGTERM. Shutting down')
+        this.terminate()
+      } else {
+        debug('Got SIGTERM. But did not yet see proxy terminating, deferring shutdown until proxy stops')
+        // no timeout because process will be SIGKILLed anyway if it does not get the proxy termination in time
```

Comment:
Was wondering about a timeout, but makes sense that it will be killed eventually anyway.